projects/step-frontend/src/lib/modules/timeseries/chart/time-series-chart.component.ts

```diff
@@ -57,6 +57,13 @@ export class TimeSeriesChartComponent implements OnInit, AfterViewInit, OnChange
     }
   }
 
+  setTitle(title: string): void {
+    let titles = this.chartElement.nativeElement.getElementsByClassName('u-title');
+    if (titles.length) {
+      titles[0].innerHTML = title;
+    }
+  }
```

Comment:
In general we shouldn't need this kind of programatical way of setting things but use inputs in the template

Reply:
I checked the entire library for ability to change the title dinamically, but it does not have it. This is the only option in this case since I'm manipulating directly elements generated by uPlot.